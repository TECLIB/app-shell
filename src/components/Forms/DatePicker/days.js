import React from 'react'

/**
 * List of days depending on the month and the year
 * @function days
 * @return {component} List of options for a select
 */
export default (year, month) => {
  const daysList = {
    January: 31,
    February: 28,
    March: 31,
    April: 30,
    May: 31,
    June: 30,
    July: 31,
    August: 31,
    September: 30,
    October: 31,
    November: 30,
    December: 31,
  }

  if (year % 400 === 0 || (year % 100 !== 0 && year % 4 === 0)) {
    daysList.February = 29
  }

  const days = []

  if (month) {
    for (let index = 1; index <= daysList[month]; index += 1) {
      days.push((
        <option value={index} key={`day-${index.toString()}`}>
          { index }
        </option>
      ))
    }
  }

  return days
}
