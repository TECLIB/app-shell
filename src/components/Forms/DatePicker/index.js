import React, {
  PureComponent,
} from 'react'
import PropTypes from 'prop-types'
import YEARS from './years'
import MONTHS from './months'
import DAYS from './days'
import monthsList from './monthsList.json'

/**
 * Component to create a date picker
 * @class DatePicker
 * @extends PureComponent
 */
class DatePicker extends PureComponent {
  /** @constructor */
  constructor(props) {
    super(props)
    const date = new Date(this.props.value)
    this.state = {
      year: this.props.value ? date.getFullYear() : undefined,
      month: this.props.value ? monthsList[date.getMonth()] : undefined,
      day: this.props.value ? date.getDate() : undefined,
    }
  }

  /**
   * Return the name and value to the father
   * @function change
   * @param {object} eventObject
   */
  change = (eventObject) => {
    const currentDate = new Date()
    const newDate = {
      year: this.state.year ? this.state.year : currentDate.getFullYear(),
      month: this.state.month ? this.state.month : monthsList[currentDate.getMonth()],
      day: this.state.day ? this.state.day : 1,
    }

    newDate[eventObject.target.name] = eventObject.target.value

    this.setState({ ...newDate },
      this.props.function(this.props.name, new Date(`${newDate.year} ${newDate.month} ${newDate.day}`)))
  }

  /**
   * Render component
   * @function render
   */
  render() {
    return (
      <div className="froms__col">
        <p>
          {this.props.label}
        </p>

        <div className="win-disposable win-datepicker">
          <select
            required={this.props.required}
            className="win-datepicker-month win-order0"
            name="month"
            value={this.state.month}
            onChange={this.change}
          >
            <option>
              ---
            </option>
            { MONTHS() }
          </select>

          <select
            className="win-datepicker-date win-order1"
            name="day"
            value={this.state.day}
            onChange={this.change}
          >
            <option>
              ---
            </option>
            { DAYS(this.state.year, this.state.month) }
          </select>

          <select
            className="win-datepicker-year win-order2"
            name="year"
            value={this.state.year}
            onChange={this.change}
          >
            <option>
              ---
            </option>
            { YEARS() }
          </select>
        </div>
      </div>
    )
  }
}

DatePicker.defaultProps = {
  required: false,
  value: undefined,
}

DatePicker.propTypes = {
  label: PropTypes.string.isRequired,
  name: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]).isRequired,
  value: PropTypes.instanceOf(Date),
  function: PropTypes.func.isRequired,
  required: PropTypes.bool,
}

export default DatePicker
