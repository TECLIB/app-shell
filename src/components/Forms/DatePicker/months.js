import React from 'react'
import monthsList from './monthsList.json'

/**
 * List months of the year
 * @function months
 * @return {component} List of options for a select
 */
export default () => {
  const months = []
  monthsList.forEach((month) => {
    months.push((
      <option value={month} key={`month${month}`}>
        { month }
      </option>
    ))
  })
  return months
}
