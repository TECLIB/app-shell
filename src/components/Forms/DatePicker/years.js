import React from 'react'

/**
 * List of years between 1980 and 2100
 * @function years
 * @return {component} List of options for a select
 */
export default () => {
  const years = []
  for (let index = 1980; index <= 2100; index += 1) {
    years.push(
      <option value={index} key={`year${index}`}>
        { index }
      </option>,
    )
  }
  return years
}
