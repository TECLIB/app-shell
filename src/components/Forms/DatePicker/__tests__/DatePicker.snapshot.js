import React from 'react'
import renderer from 'react-test-renderer'
import DatePicker from '../index'

describe('DatePicker', () => {
  test('renders DatePicker', () => {
    const component = renderer.create(
      <DatePicker
        name="test"
        label="test"
        value={new Date('06-10-1997')}
        function={() => {}}
        required
      />,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
