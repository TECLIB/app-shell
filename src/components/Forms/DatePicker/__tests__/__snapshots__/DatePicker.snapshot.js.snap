// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker renders DatePicker 1`] = `
<div
  className="froms__col"
>
  <p>
    test
  </p>
  <div
    className="win-disposable win-datepicker"
  >
    <select
      className="win-datepicker-month win-order0"
      name="month"
      onChange={[Function]}
      required={true}
      value="June"
    >
      <option>
        ---
      </option>
      <option
        value="January"
      >
        January
      </option>
      <option
        value="February"
      >
        February
      </option>
      <option
        value="March"
      >
        March
      </option>
      <option
        value="April"
      >
        April
      </option>
      <option
        value="May"
      >
        May
      </option>
      <option
        value="June"
      >
        June
      </option>
      <option
        value="July"
      >
        July
      </option>
      <option
        value="August"
      >
        August
      </option>
      <option
        value="September"
      >
        September
      </option>
      <option
        value="October"
      >
        October
      </option>
      <option
        value="November"
      >
        November
      </option>
      <option
        value="December"
      >
        December
      </option>
    </select>
    <select
      className="win-datepicker-date win-order1"
      name="day"
      onChange={[Function]}
      value={10}
    >
      <option>
        ---
      </option>
      <option
        value={1}
      >
        1
      </option>
      <option
        value={2}
      >
        2
      </option>
      <option
        value={3}
      >
        3
      </option>
      <option
        value={4}
      >
        4
      </option>
      <option
        value={5}
      >
        5
      </option>
      <option
        value={6}
      >
        6
      </option>
      <option
        value={7}
      >
        7
      </option>
      <option
        value={8}
      >
        8
      </option>
      <option
        value={9}
      >
        9
      </option>
      <option
        value={10}
      >
        10
      </option>
      <option
        value={11}
      >
        11
      </option>
      <option
        value={12}
      >
        12
      </option>
      <option
        value={13}
      >
        13
      </option>
      <option
        value={14}
      >
        14
      </option>
      <option
        value={15}
      >
        15
      </option>
      <option
        value={16}
      >
        16
      </option>
      <option
        value={17}
      >
        17
      </option>
      <option
        value={18}
      >
        18
      </option>
      <option
        value={19}
      >
        19
      </option>
      <option
        value={20}
      >
        20
      </option>
      <option
        value={21}
      >
        21
      </option>
      <option
        value={22}
      >
        22
      </option>
      <option
        value={23}
      >
        23
      </option>
      <option
        value={24}
      >
        24
      </option>
      <option
        value={25}
      >
        25
      </option>
      <option
        value={26}
      >
        26
      </option>
      <option
        value={27}
      >
        27
      </option>
      <option
        value={28}
      >
        28
      </option>
      <option
        value={29}
      >
        29
      </option>
      <option
        value={30}
      >
        30
      </option>
    </select>
    <select
      className="win-datepicker-year win-order2"
      name="year"
      onChange={[Function]}
      value={1997}
    >
      <option>
        ---
      </option>
      <option
        value={1980}
      >
        1980
      </option>
      <option
        value={1981}
      >
        1981
      </option>
      <option
        value={1982}
      >
        1982
      </option>
      <option
        value={1983}
      >
        1983
      </option>
      <option
        value={1984}
      >
        1984
      </option>
      <option
        value={1985}
      >
        1985
      </option>
      <option
        value={1986}
      >
        1986
      </option>
      <option
        value={1987}
      >
        1987
      </option>
      <option
        value={1988}
      >
        1988
      </option>
      <option
        value={1989}
      >
        1989
      </option>
      <option
        value={1990}
      >
        1990
      </option>
      <option
        value={1991}
      >
        1991
      </option>
      <option
        value={1992}
      >
        1992
      </option>
      <option
        value={1993}
      >
        1993
      </option>
      <option
        value={1994}
      >
        1994
      </option>
      <option
        value={1995}
      >
        1995
      </option>
      <option
        value={1996}
      >
        1996
      </option>
      <option
        value={1997}
      >
        1997
      </option>
      <option
        value={1998}
      >
        1998
      </option>
      <option
        value={1999}
      >
        1999
      </option>
      <option
        value={2000}
      >
        2000
      </option>
      <option
        value={2001}
      >
        2001
      </option>
      <option
        value={2002}
      >
        2002
      </option>
      <option
        value={2003}
      >
        2003
      </option>
      <option
        value={2004}
      >
        2004
      </option>
      <option
        value={2005}
      >
        2005
      </option>
      <option
        value={2006}
      >
        2006
      </option>
      <option
        value={2007}
      >
        2007
      </option>
      <option
        value={2008}
      >
        2008
      </option>
      <option
        value={2009}
      >
        2009
      </option>
      <option
        value={2010}
      >
        2010
      </option>
      <option
        value={2011}
      >
        2011
      </option>
      <option
        value={2012}
      >
        2012
      </option>
      <option
        value={2013}
      >
        2013
      </option>
      <option
        value={2014}
      >
        2014
      </option>
      <option
        value={2015}
      >
        2015
      </option>
      <option
        value={2016}
      >
        2016
      </option>
      <option
        value={2017}
      >
        2017
      </option>
      <option
        value={2018}
      >
        2018
      </option>
      <option
        value={2019}
      >
        2019
      </option>
      <option
        value={2020}
      >
        2020
      </option>
      <option
        value={2021}
      >
        2021
      </option>
      <option
        value={2022}
      >
        2022
      </option>
      <option
        value={2023}
      >
        2023
      </option>
      <option
        value={2024}
      >
        2024
      </option>
      <option
        value={2025}
      >
        2025
      </option>
      <option
        value={2026}
      >
        2026
      </option>
      <option
        value={2027}
      >
        2027
      </option>
      <option
        value={2028}
      >
        2028
      </option>
      <option
        value={2029}
      >
        2029
      </option>
      <option
        value={2030}
      >
        2030
      </option>
      <option
        value={2031}
      >
        2031
      </option>
      <option
        value={2032}
      >
        2032
      </option>
      <option
        value={2033}
      >
        2033
      </option>
      <option
        value={2034}
      >
        2034
      </option>
      <option
        value={2035}
      >
        2035
      </option>
      <option
        value={2036}
      >
        2036
      </option>
      <option
        value={2037}
      >
        2037
      </option>
      <option
        value={2038}
      >
        2038
      </option>
      <option
        value={2039}
      >
        2039
      </option>
      <option
        value={2040}
      >
        2040
      </option>
      <option
        value={2041}
      >
        2041
      </option>
      <option
        value={2042}
      >
        2042
      </option>
      <option
        value={2043}
      >
        2043
      </option>
      <option
        value={2044}
      >
        2044
      </option>
      <option
        value={2045}
      >
        2045
      </option>
      <option
        value={2046}
      >
        2046
      </option>
      <option
        value={2047}
      >
        2047
      </option>
      <option
        value={2048}
      >
        2048
      </option>
      <option
        value={2049}
      >
        2049
      </option>
      <option
        value={2050}
      >
        2050
      </option>
      <option
        value={2051}
      >
        2051
      </option>
      <option
        value={2052}
      >
        2052
      </option>
      <option
        value={2053}
      >
        2053
      </option>
      <option
        value={2054}
      >
        2054
      </option>
      <option
        value={2055}
      >
        2055
      </option>
      <option
        value={2056}
      >
        2056
      </option>
      <option
        value={2057}
      >
        2057
      </option>
      <option
        value={2058}
      >
        2058
      </option>
      <option
        value={2059}
      >
        2059
      </option>
      <option
        value={2060}
      >
        2060
      </option>
      <option
        value={2061}
      >
        2061
      </option>
      <option
        value={2062}
      >
        2062
      </option>
      <option
        value={2063}
      >
        2063
      </option>
      <option
        value={2064}
      >
        2064
      </option>
      <option
        value={2065}
      >
        2065
      </option>
      <option
        value={2066}
      >
        2066
      </option>
      <option
        value={2067}
      >
        2067
      </option>
      <option
        value={2068}
      >
        2068
      </option>
      <option
        value={2069}
      >
        2069
      </option>
      <option
        value={2070}
      >
        2070
      </option>
      <option
        value={2071}
      >
        2071
      </option>
      <option
        value={2072}
      >
        2072
      </option>
      <option
        value={2073}
      >
        2073
      </option>
      <option
        value={2074}
      >
        2074
      </option>
      <option
        value={2075}
      >
        2075
      </option>
      <option
        value={2076}
      >
        2076
      </option>
      <option
        value={2077}
      >
        2077
      </option>
      <option
        value={2078}
      >
        2078
      </option>
      <option
        value={2079}
      >
        2079
      </option>
      <option
        value={2080}
      >
        2080
      </option>
      <option
        value={2081}
      >
        2081
      </option>
      <option
        value={2082}
      >
        2082
      </option>
      <option
        value={2083}
      >
        2083
      </option>
      <option
        value={2084}
      >
        2084
      </option>
      <option
        value={2085}
      >
        2085
      </option>
      <option
        value={2086}
      >
        2086
      </option>
      <option
        value={2087}
      >
        2087
      </option>
      <option
        value={2088}
      >
        2088
      </option>
      <option
        value={2089}
      >
        2089
      </option>
      <option
        value={2090}
      >
        2090
      </option>
      <option
        value={2091}
      >
        2091
      </option>
      <option
        value={2092}
      >
        2092
      </option>
      <option
        value={2093}
      >
        2093
      </option>
      <option
        value={2094}
      >
        2094
      </option>
      <option
        value={2095}
      >
        2095
      </option>
      <option
        value={2096}
      >
        2096
      </option>
      <option
        value={2097}
      >
        2097
      </option>
      <option
        value={2098}
      >
        2098
      </option>
      <option
        value={2099}
      >
        2099
      </option>
      <option
        value={2100}
      >
        2100
      </option>
    </select>
  </div>
</div>
`;
